changelog-tool/changelog.pony

```diff
@@ -79,25 +85,49 @@ class Release
     added = Section._empty(Added)
     changed = Section._empty(Changed)
 
+  fun ref add_entry(section_name: String, entry: String) ? =>
+    let section =
+      match section_name
+      | "fixed" =>
+        if fixed is None then fixed = Section._empty(Fixed) end
+        fixed as Section
+      | "added" =>
+        if added is None then added = Section._empty(Added) end
+        added as Section
+      | "changed" =>
+        if changed is None then changed = Section._empty(Changed) end
+        changed as Section
+      else error
+      end
+    section.entries.push("- " + entry)
+
   fun string(): String iso^ =>
-    if heading == _unreleased_heading then
-      "\n\n".join(
-        [ heading
-          "### Fixed\n"
-          "### Added\n"
-          "### Changed\n"
-          ""
-        ].values())
-    else
-      let str = recover String .> append(heading) .> append("\n\n") end
-      for section in [fixed; added; changed].values() do
-        match section
-        | let s: Section box =>
-          str .> append(s.string()) .> append("\n")
-        end
+    // In order to represent the empty sections of unreleased releases,
+    // we must use the empty correspoding section when printing instead
+    // of None, otherwise it will be ignored.
+    let fixed' = match fixed
```

Comment:
These match expressions should begin on the following line with one level of indentation: https://github.com/ponylang/ponyc/blob/master/STYLE_GUIDE.md#assignment

Reply:
Argh, pardon my ignorance!

Reply:
No worries, our style takes some time to get used to.